source/common/router/header_formatter.cc

```diff
@@ -179,21 +179,23 @@ parsePerRequestStateField(absl::string_view param_str) {
   return [param](const Envoy::StreamInfo::StreamInfo& stream_info) -> std::string {
     const Envoy::StreamInfo::FilterState& filter_state = stream_info.filterState();
 
+    auto state = filter_state.getDataReadOnlyGeneric(param);
     // No such value means don't output anything.
-    if (!filter_state.hasDataWithName(param)) {
+    if (state == nullptr) {
       return std::string();
     }
 
+    auto typed_state = dynamic_cast<const StringAccessor*>(state);
```

Comment:
Per other comment dynamic_cast on nullptr is OK so you can collapse this and make it simpler. Same elsewhere.

Reply:
get it.